recipes/avahi/all/conanfile.py

```diff
@@ -26,75 +29,67 @@ class AvahiConan(ConanFile):
+                pass
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    @property
-    def _configure_args(self):
-        yes_no = lambda v: "yes" if v else "no"
-        return [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-            "--disable-gtk3",
-            "--disable-mono",
-            "--disable-python",
-            "--disable-qt5",
-            "--disable-monodoc",
-            "--enable-compat-libdns_sd",
-            "--with-systemdsystemunitdir={}/lib/systemd/system".format(self.package_folder),
-        ]
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        self._autotools.configure(configure_dir=self._source_subfolder, args=self._configure_args)
-        return self._autotools
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.append("--enable-compat-libdns_sd")
+        tc.configure_args.append("--disable-gtk3")
+        tc.configure_args.append("--disable-mono")
+        tc.configure_args.append("--disable-monodoc")
+        tc.configure_args.append("--disable-python")
+        tc.configure_args.append("--disable-qt5")
+        tc.configure_args.append("--with-systemdsystemunitdir=/lib/systemd/system")
+        tc.generate()
+        AutotoolsDeps(self).generate()
+        PkgConfigDeps(self).generate()
 
     def build(self):
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
+        autotools.configure()
         autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
+        autotools = Autotools(self)
         autotools.install()
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        tools.rmdir(os.path.join(self.package_folder, "etc"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        rmdir(self, os.path.join(self.package_folder, "etc"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "run"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "Avahi"
-        self.cpp_info.names["cmake_find_package_multi"] = "Avahi"
-
```

Comment:
@jwillikers Why were these removed?
The target names used in the original recipe were based on some common practice, e.g.
* https://github.com/sony/nmos-cpp/blob/master/Development/third_party/cmake/FindAvahi.cmake
* https://github.com/ruslo/hunter/blob/master/cmake/find/FindAvahi.cmake

Reply:
Conan Center's policy, documented [here](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/reviewing.md#cmake-targets), is to match target names of either those provided by CMake's own Find Modules or those officially provided by the library itself. When adding support for the newer Conan CMakeDeps generator, this was brought up. Instead of making the names inconsistent by following the policy for the CMakeDeps generator, it was determined to remove the previous aliases. You might try opening an issue in Conan Center to get feedback from the other Conan Center contributors on this.

Reply:
Thanks, I really appreciate the quick reply. I opened #14223 but I have little hope of it being merged, despite that the uppercase named target `Avahi::Avahi` is common practice since at least 2015, because no-one who adopted it would be considered "official".